src/irmin-tezos-utils/ring.ml

```diff
@@ -0,0 +1,21 @@
+type 'a t = { n : int; mutable i : int; a : 'a array }
+
+let make n ~default = { n; i = 0; a = Array.make n default }
```

Comment:
I'm not sure I like the handling of `option` for entries below, would it make sense to have `'a option array`? :) (and guarantee that values between `0..i` are all `Some` and the rest is `None`?)

Reply:
Indeed, i'll do that